py/dml/serialize.py

```diff
@@ -277,6 +301,11 @@ def map_dmltype_to_attrtype(site, dmltype):
         return '[%s{%s}]' % (arr_attr_type, arr_length) + or_data
     if isinstance(real_type, TObjIdentity):
         return '[s[i*]]'
+    if isinstance(real_type, TTrait):
+        serialized_traits.add(real_type.trait)
+        # TODO Is this redundant?
+        real_type.trait.mark_referenced()
```

Comment:
Yes, in the sense that we should see if removing trait refcounting incurs a cost.